app/controllers/products_controller.rb

```diff
@@ -3,6 +3,10 @@ class ProductsController < ApplicationController
   before_action :set_shop
   before_action :set_product, only: [:edit, :update, :destroy]
 
+  add_breadcrumb "Products", :shop_products_path
+
+  add_breadcrumb "New Product", :new_shop_product_path
```

Comment:
It is on the new method. The breadcrumb method looks at the path and determines the action from that.